libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/BotFrameworkServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Bot.Builder;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Bot.Builder.Integration.AspNet.Core
+{
+    /// <summary>
+    /// An extention to the Service Collection to allow registration of the Bot Framework Protocol Version 3 with Streaming Extensions adapter for use with named pipe connections.
+    /// </summary>
+    public static partial class BotFrameworkServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Used in the Configuration section of Startup to register the adapter for use with named pipe connections.
+        /// Throws <see cref="ArgumentNullException"/> if services is null.
+        /// </summary>
+        /// <param name="services">The current service collection, passed in here to enable use of the adapter.</param>
+        /// <returns>The service collection, updated to include the new adapter.</returns>
+        public static IServiceCollection AddNamedPipeConnector(this IServiceCollection services)
```

Comment:
>public static IServiceCollection AddNamedPipeConnector(this IServiceCollection services) [](start = 8, length = 88)

I think this can be marked internal